torch_xla/utils/checkpoint_tagger.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import division
+from __future__ import print_function
+
+from collections import defaultdict
+import json
+
+class CheckpointTagger(object):
+
+  def __init__(self, remover=None):
+    self._tags = dict()
+    self._refcount = defaultdict(int)
+    remover = (lambda x: None) if remover is None else remover
+    assert callable(remover)
+    self._remover = remover
+
+  def tag(self, name, path):
+    self._refcount[path] += 1
+    old_path = self._tags.get(name)
+    if old_path is not None:
+      self._refcount[old_path] -= 1
+      if self._refcount[old_path] == 0:
+        self._refcount.pop(old_path)
+        self._remover(old_path)
+    self._tags[name] = path
+
+  def dump(self):
+    return json.dumps(self._tags)
+
+  @classmethod
+  def load(cls, dat, remover=None):
```

Comment:
Please simplify all the logic.
Input is string JSON.

Rename in load_from_json().

Reply:
done